Update Makefile.riscv64, remove --fast-math from default build options on x280

fleclairTT opened the pull request:
The default flags for various targets should assume compliance with IEEE-754. fast math and co should be opt in only

fleclairTT committed:
Update Makefile.riscv64, remove --fast-math from default build options

The default flags for various targets should assume compliance with IEEE-754. fast math and co should be opt in only

martin-frbg commented:
I had wondered about this one too - IIRC the -ffast-math was in the initial code drop on the riscv branch, but whether it was there to paper over some compiler issue or just out of habit of the author I do not know @HellerZheng

fleclairTT committed:
add general compiler target for tt-ascalon-d8

ascalon d8 is the tenstorrent ascalon processor in it's 8 wide configuration featuring multiple RVV1.0 compliant, 256 bit wide vector units. 

It also support a number of optional extensions for RVV, inclusive of: Ascalon Vector unit is designed to enable V-extension with a handful of minor extensions required or optional of the RVA23 Profile.
V: Vector Extension for Application Processors
Zvfh: Vector Extension for Half-Precision Floating-Point
Zvkn: NIST Algorithm Suite
Zvkg: Vector GCM/GMAC
Zvbc: Vector Carryless Multiplication
Zvbb: Vector Basic Bit-manipulation
Zvfbfwma: Vector BF16 widening mul-add
Xsfvqdotq: Vector quad widening 4D Dot Product

martin-frbg reviewed (commented):
wait, that won't work without all the other infrastructure changes for a new target - KERNEL file, param.h entries for GEMM block sizes etc. Better make that a separate PR

fleclairTT commented:
Yeah, the second PR wasn't supposed to be tracked here, apologies

HellerZheng commented:
> I had wondered about this one too - IIRC the -ffast-math was in the initial code drop on the riscv branch, but whether it was there to paper over some compiler issue or just out of habit of the author I do not know @HellerZheng
Sure, please go ahead.

fleclairTT commented:
resolved by https://github.com/OpenMathLib/OpenBLAS/pull/5055
